components/layout_2020/taffy/layout.rs

```diff
@@ -108,6 +110,14 @@ impl taffy::LayoutPartialTree for TaffyContainerContext<'_> {
         (*self.source_child_nodes[id]).borrow_mut().taffy_layout = *layout;
     }
 
+    fn resolve_calc_value(&self, val: u64, basis: f32) -> f32 {
+        let calc_ptr = val as usize as *const CalcLengthPercentage;
```

Comment:
This interface (passing pointers through integers) makes tracking pointer provenance a bit more tricky for tools like MIRI (see [Exposed Provenance](https://doc.rust-lang.org/nightly/std/ptr/index.html#exposed-provenance)).

I am not sure if there are goals in running things through miri to check for various kinds of UB, but this will certainly make it harder to track "things"...

I guess one simple alternative would be to change the interface to something like:

```rust
union ResolveCalcValue {
    ptr: *const (),
    val: u64,
}

// or 

enum ResolveCalcValue {
    Ptr(NonNull<()>),
    Val(u64),
}

fn resolve_calc_value(&self, val: ResolveCalcValue, basis: f32) -> f32 {...}
```

Reply:
I [asked about this](https://rust-lang.zulipchat.com/#narrow/channel/136281-t-opsem/topic/Strict.20provenance.3A.20how.20to.20deal.20with.20.22maybe.20pointer.22.20APIs). Conclusion was that if it is just a `usize`-sized value that needs to be supported then one can just use `*const ()`. If `u64`-sized value is required then a `union` is the way to go.

I don't think an enum will work because we need to store this in a maximum of 64 bits and so we won't know which variant we have when we pass it back.

Reply:
I have updated the Taffy API (and internal implementation) to use `*const ()`